src/caffeine/collective_subroutines_s.f90

```diff
@@ -10,30 +11,306 @@
 
    !! void c_co_sum_no_result_image_int32(void* c_loc_a, int Nelem)
    interface
+
      subroutine c_co_sum_no_result_image_int32(c_loc_a, Nelem) bind(C)
-       import c_int32_t, c_ptr, c_size_t
+       import c_ptr, c_size_t
+       type(c_ptr), value :: c_loc_a
+       integer(c_size_t), value :: Nelem
+     end subroutine
+
+     subroutine c_co_sum_no_result_image_int64(c_loc_a, Nelem) bind(C)
+       import c_ptr, c_size_t
+       type(c_ptr), value :: c_loc_a
+       integer(c_size_t), value :: Nelem
+     end subroutine
+
+     subroutine c_co_sum_no_result_image_float(c_loc_a, Nelem) bind(C)
+       import c_ptr, c_size_t
+       type(c_ptr), value :: c_loc_a
+       integer(c_size_t), value :: Nelem
+     end subroutine
+
+     subroutine c_co_sum_no_result_image_double(c_loc_a, Nelem) bind(C)
+       import c_ptr, c_size_t
        type(c_ptr), value :: c_loc_a
        integer(c_size_t), value :: Nelem
      end subroutine
+
    end interface
 
    select rank(a)
      rank(0) 
        select type(a)
          type is(integer(c_int32_t))
            call c_co_sum_no_result_image_int32(c_loc(a), nelem=1_c_size_t)
+         type is(integer(c_int64_t))
+           call c_co_sum_no_result_image_int64(c_loc(a), nelem=1_c_size_t)
+         type is(real(c_float))
+           call c_co_sum_no_result_image_float(c_loc(a), nelem=1_c_size_t)
+         type is(real(c_double))
+           call c_co_sum_no_result_image_double(c_loc(a), nelem=1_c_size_t)
+         type is(complex(c_float))
+           call c_co_sum_no_result_image_float(c_loc(a), nelem=2_c_size_t)
+         type is(complex(c_double))
+           call c_co_sum_no_result_image_double(c_loc(a), nelem=2_c_size_t)
```

Comment:
Forgive my ignorance, where are 1_c_size_t and 2_c_size_t defined ?

Reply:
These are what Fortran calls literal constants so they are just numbers.   The  `_c_size_t` suffix tells the Fortran compiler that I want the preceding number to be stored in a bit representation that exactly matches the companion C compiler's `size_t` representation.  According to the Fortran standard, every Fortran compiler must have a companion C compiler.  In most cases, it's the C compiler from the same vendor with the exception being vendors that don't produce a C compiler and therefore designate a third-party C compiler as the companion C compiler.